### PR TITLE
Update metadata validation logic (issue #924)

### PR DESCRIPTION
**Pull request by Iinh**
- Issue #924
- What I did:
    - refactor parse_metadata with `attr` validation
    - add file check for `is_metadata_file()`
    - update metadata tests
- Looking forward to review and feedback @scholtzan. Thank you!

### CODE REVIEW

File: bigquery_etl/metadata/parse_metadata.py [hunk 1]
```diff
@@ -42,8 +72,10 @@ def is_metadata_file(file_path):
 
         Checks if the name and file format match the metadata file requirements.
         """
-        # todo: we should probably also check if the file actually exists etc.
-        return os.path.basename(file_path) == METADATA_FILE
```
Comment (Iinh) [inline bigquery_etl/metadata/parse_metadata.py:46]: I updated this method to include file checking per the TODO, but CircleCI failed it. Should I revert this change or is there a way to update the CircleCI test?

**Comment by Iinh** [comment]
Hi @scholtzan. I'm trying to update my PR, is there a script to automatically lint my code? I know there is an `npm run format` in the Glean repo, I was wondering if we have something equivalent for this repo. Thank you!

**Comment by scholtzan** [comment]
Yes, there is something similar here. You can run `black bigquery_etl && flake8 bigquery_etl && pydocstyle bigquery_etl` which will automatically format your code and provide information about other issues (unused imports, malformed doc strings, ...)

**Comment by Iinh** [comment]
Hi @scholtzan I removed all the format errors. Could you help review if I need to add/fix anything? Thank you.

**Review by scholtzan — APPROVED** [review]
This looks great! :) Thank you for taking this on!
